src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,150 @@
+package duke;
+
+import command.*;
+import duke_exception.*;
+
+/**
+ * Parser handles the interpretation of user input,
+ * and makes the appropriate method calls.
+ *
+ * @author Ho Wen Zhong
+ */
+public class Parser {
+
+    /**
+     * Returns the appropriate Command based on user input.
+     *
+     * @param fullCommand User input.
+     * @return The appropriate Command.
+     * @throws DukeException If there are syntax errors in user input.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        try {
+            if (fullCommand.equals("bye")) {
+                return new ExitCommand();
+            } else if (fullCommand.equals("list")) {
+                return new ListCommand();
+            } else if (fullCommand.startsWith("done")) {
+                return detectDone(fullCommand);
+            } else if (fullCommand.startsWith("todo")) {
+                return detectTodo(fullCommand);
+            } else if (fullCommand.startsWith("deadline")) {
+                return detectDeadline(fullCommand);
+            } else if (fullCommand.startsWith("event")) {
+                return detectEvent(fullCommand);
+            } else if (fullCommand.startsWith("delete")) {
+                return detectDelete(fullCommand);
+            } else if (fullCommand.startsWith("find")){
+                return detectFind(fullCommand);
```

Comment:
Function naming starting with **detect** make it sound like it has a boolean return value rather than a Command. May be try something like createDoneCommand(fullCommand).